webots_ros2_tutorials/README.md

```diff
@@ -0,0 +1,68 @@
+<p align="center"><img src="images/gsod_06.png" alt="Video 6" height="300"></p>
+
+### Build the project using:
+`colcon build`
+### Launch the project using: 
+``ros2 launch webots_ros2_tutorial line_following.launch``
+
+``ros2 topic list`` shows the list of all topics. 
+
+``ros2 topic echo /topic_name`` can be used to subscribe to an topic.
+
+# [Implementation of SLAM toolbox in an unknown environment](https://youtu.be/3Ak3KoNhfv0)
+This includes the [10th video](https://youtu.be/3Ak3KoNhfv0) and the [11th video](https://youtu.be/3Ak3KoNhfv0) in the [Webots-ROS YouTube Tutorial Series](https://www.youtube.com/watch?v=jU_FD1_zAqo&list=PLt69C9MnPchkP0ZXZOqmIGRTOch8o9GiQ)
+This [SteveMacenski's SLAM toolbox github repo](https://github.com/SteveMacenski/slam_toolbox) is used to study about the SLAM toolbox package. It includes different services and plugins for Rviz2 for working with this package.
+
+### Command to install SLAM toolbox :
+```
+apt install ros-foxy-slam-toolbox
+```
+This project based on SLAM generates a final map in RVIZ which matches that of the Webots world. It gives the viewers a god understanding of concepts such as odometry, localization and mapping. Here the robot is navigated using keyboard or joystick commands. Once the robots starts to move, it’s scan and odometry is taken by the slam node and a map is published which can be seen in rviz2. The project algorithm also shifts odom with respect to map in order of match the scan with the map. Hence we get a consistent map. A lidar is added to the custom robot so that it is able to publish the scan. Other sensors like distance sensor and wheels made use of in the project have been described and enabled in the videos. The need to publish odometry and transform between odom and base_link in order to use SLAM toolbox to generate and correct the map has also been explained. The master node used here is similar to the one used in the Line Following Robot project. It is used to generate cmd_vel depending on the readings of the sensor. SLAM configuration is performed in order to facilitate publishing and update of the map and accordingly change transform between map and odom link by matching the laser scan with the generated occupancy grip. Hence it truly does localization at each step before adding points in the occupancy grid that are mapping. ROS2 framework is used and hence we see how to setup different world files, protos, and launch file in the setup.py. The final launch file has 2 parts: In the first part we use an inbuilt webots framework to call the world and call files to enable all the sensors and wheels in the custom robot. In the second part we make a node for line following logic. This node takes in IR sensor readings and processes the data. Finally it spits out cmd_vel which can be used by robot for navigation.
```

Comment:
```suggestion
This project based on SLAM generates a final map in RViz2 which matches that of the Webots world. It gives the viewers a god understanding of concepts such as odometry, localization and mapping. Here the robot is navigated using keyboard or joystick commands. Once the robots starts to move, it’s scan and odometry is taken by the slam node and a map is published which can be seen in RViz2. The project algorithm also shifts odometry with respect to map in order of match the scan with the map. Hence we get a consistent map. A lidar is added to the custom robot so that it is able to publish the scan. Other sensors like distance sensor and wheels made use of in the project have been described and enabled in the videos. The need to publish odometry and transform between odom and base_link in order to use SLAM toolbox to generate and correct the map has also been explained. The master node used here is similar to the one used in the Line Following Robot project. It is used to generate `cmd_vel` depending on the readings of the sensor. SLAM configuration is performed in order to facilitate publishing and update of the map and accordingly change transform between `map` and `odom` link by matching the laser scan with the generated occupancy grip. Hence it truly does localization at each step before adding points in the occupancy grid that are mapping. ROS2 framework is used and hence we see how to setup different world files, protos, and launch file in the setup.py. The final launch file has 2 parts: In the first part we use an inbuilt Webots framework to call the world and call files to enable all the sensors and wheels in the custom robot. In the second part we make a node for line following logic. This node takes in IR sensor readings and processes the data. Finally it spits out `cmd_vel` which can be used by robot for navigation.
```